lib/smart_proxy_openscap/openscap_lib.rb

```diff
@@ -57,7 +57,14 @@ def self.common_name(request)
   def self.send_spool_to_foreman(loaded_settings)
     arf_dir = File.join(loaded_settings.spooldir, "/arf")
     return unless File.exist? arf_dir
+    lockfile = File.join(loaded_settings.spooldir, ".spool.lock")
+    if File.exist? lockfile
+      logger.debug "Lock file #{lockfile} for openscap spool exists, not sending spool to server"
+      return
+    end
+    FileUtils.touch lockfile
     SpoolForwarder.new(loaded_settings).post_arf_from_spool(arf_dir)
+    FileUtils.rm lockfile
```

Comment:
Can you put the removal to `ensure` block? We should probably also install SIGTERM trap to remove it as well.